spec/models/rating_spec.rb

```diff
@@ -1,25 +1,49 @@
-require_relative '../../app/models/rating'
+require 'rails_helper'
 
-describe Rating do
-  before do
-    @rating = Rating.new
-  end
+RSpec.describe Rating, type: :model do
+  context 'rating in relation to taco' do
+    subject do
+      ingredients = {
+        shell: Shell.new,
+        meat: Meat.new,
+        veg: Veggie.new
+      }
+      Taco.create(Rating.new, ingredients)
+    end
 
-  it 'can instantiate' do
-    @rating.nil?
-  end
+    it 'initializes to nil' do
+      expect(subject.rating).to eq(nil)
+    end
 
-  it 'defaults to nil before first rating' do
-    @rating.get_rating.nil?
-  end
+    it 'will become 5 when rated up 5 once' do
+      subject.rate(5)
+      expect(subject.rating).to eq(5)
+    end
 
-  it 'has zero ratings to start' do
-    @rating.get_num_ratings == 0
+    it 'will average and become 3 when second rating is 1' do
+      subject.rate(1)
+      expect(subject.rating).to eq(3)
+    end
   end
 
-  it 'can update rating' do
-    @rating.rate_up(5)
-    @rating.get_rating == 5
-    @rating.get_num_ratings == 1
+  context 'rating in relation to taco shop' do
+    subject do
+      cheesy = Taco.new(Rating.new(3.5))
+      meaty = Taco.new(Rating.new(4.5))
+      TacoShop.create({cheesy: cheesy, meaty: meaty})
+    end
+    
+    it 'instantiates when given tacos with ratings' do
+      expect(subject.rating).to not be_nil
+    end
+
+    it 'averages ratings if multiple tacos' do
+      expect(subject.rating).to eq(4) # (3.5 + 4.5) / 2
+    end
+    
+    it 'assigns same rating as taco if shop has only one taco' do
+      subject.remove_taco(:meaty)
+      expect(subject.rating).to eq(3.5)
```

Comment:
I thought you may want to have the subject be the rating since that's what the expectations are built off of, but then you'd also need to define a let of tacoshop so that you could still access the `remove_taco` method for this test.